index.d.ts

```diff
@@ -25,4 +27,17 @@ export class AbortError extends Error {
  * 	throttled(i).then(console.log);
  * }
  */
-export default function <T extends Function>(input: T, limit: number, interval: number): T;
+export default function<TArguments extends any[], TReturn>(
```

Comment:
I prefer not prefixing types with `T`.

`any` => `unknown`

Reply:
> I prefer not prefixing types with `T`.

I'll change this, but don't you think that this actually helps distinguishing between actual types and type variables? This is such a common pattern in TS, if I'd read the types file, I'd automatically assume (without further verifying) that `Arguments` must be some declared type in the module, but `TArguments` must be a type variable.